imports/components/@enhancers/can-like/index.js

```diff
@@ -0,0 +1,90 @@
+
+import { Meteor } from "meteor/meteor";
+import React, { Component } from "react";
+import { graphql } from "react-apollo";
+import gql from "graphql-tag";
+
+import { liked as likedActions, modal } from "../../../data/store";
+import OnBoard from "../../people/accounts";
+
+const TOGGLE_LIKE_MUTATION = gql`
+  mutation ToggleLike($nodeId: String!) {
+    toggleLike(nodeId: $nodeId) {
+      like {
+        id
+      }
+    }
+  }
+`;
+
+const withToggleLike = graphql(TOGGLE_LIKE_MUTATION);
+
+/*
+  THIS NEEDS TO BE WRAPPED IN A CONNECT CALL TO WORK
+  THE CONNECT CALL MUST REVEAL
+
+  How this works:
+    it's a function.
+    inside the function, I'm declaring a class
+      * it has to be inside the function so I have access to the `WrappedComponent`
+      the class (LikesWrapper) returns the WrappedComponent with prop for toggling like (onLike)
+    at the bottom of the function I return:
+      the LikesWrapper class wrapped in a gql mutation.
+    So the export is...
+      a function that takes a React Component, and returns that component wrapped in
+      a LikesWrapper HOC which is wrapped in a GraphQl HOC that allows the toggleLike mutation
+*/
+export default (WrappedComponent) => {
```

Comment:
@JakeDawkins This is a great start. A few notes:

1.  if you use and arrow function for `toggleLike`, you don't have to bind it.
2. Currently, `getNodeId` looks over the object to try to find an `id` field, then it sets the state (thus rerendering the wrapped component.)

I think we can do this a little bit differently.

`canLike(props => props.content && props.content.id)(WrappedComponent)`

`enhancer(reducer)(Component)` is the signature.

I think we should import `react-redux` to not require the parent component to wrap with dispatch (so this enhancer is self contained).

By providing a reducer, the using component can declare where the id lives instead of it being a certain key. It also means that instead of keeping that in the enhancer state, when the like button is clicked, it calls the reducer with the most up to date props to find the id to use. This should be more efficient as it doesn't rerender the component and if the user doesn't like anything, then it never needs to call the reducer.


Does this make sense?